LEGO1/racestandsentity2.h

```diff
@@ -0,0 +1,11 @@
+#ifndef RACESTANDS2ENTITY_H
+#define RACESTANDS2ENTITY_H
+
+#include "buildingentity.h"
+
+class RaceStandsEntity2 : public BuildingEntity
```

Comment:
I think there are 3 `RaceStandsEntity`s,
with vtables at 100d48a8 100d50c0 100d5200
Looks like they don't have a common ancestor, as `GetClassName` is not shared.

Reply:
Since I also noticed that, after some thought... could they be inner classes of the respective race type they belong to? That's about the only way I can see how it makes sense they can have the same name.

Reply:
If it's an inner class, then the og code has duplicated the name string.
Perhaps this is a template class?
Time to find out what code msvc generates in this case..

Reply:
I just verified with MSVC 4.2 that the following source will emit 3 identical `name` methods.
The old linker does not have comdat folding, only `/opt:ref` (lucky for us)
https://learn.microsoft.com/en-us/cpp/build/reference/opt-optimizations

```c++
#define DLLEXPORT __declspec(dllexport)

class Core {
public:
  virtual const char *name() const {
    return "Core";
  }
};

template <class T>
class TemplateChild : public Core {
public:
  virtual const char *name() const {
    return "TemplateChild";
  }
  const T* data() const {return &m_data;}
private:
  T m_data;
};

// Create 3 exports to force the compiler to emit code for the class
DLLEXPORT TemplateChild<int> *getIntObj() {  return new TemplateChild<int>(); }
DLLEXPORT TemplateChild<float> *getFloatObj() {  return new TemplateChild<float>(); }
DLLEXPORT TemplateChild<double> *getDoubleObj() {  return new TemplateChild<double>(); }
```
Compile, and link this with:
```
cl /c a.cpp /Foa.obj
link a.obj /out:a.dll /dll /debug
```
(Looks like Ghidra does not understand the old PDB format)

Reply:
> The old linker does not have comdat folding

Oh. Oh! That's good news! That's something one always have to look out for and create rev. cludges for otherwise, not to mention it allows to know with 100% certainty that the (non-std) std collection types I have so far identified are of one single concrete type each. Thanks for the info.

(it's been about three decades since I actually used 4.2, so I'm not surprised I had forgotten these pre-std limitations and quirks)

Reply:
100D5200 JailEntity
100D50C0 CaveEntity
100D48A8 RaceStandsEntity
See my comment above on finding LegoMeterPresenter.

Reply:
100% agree!